### PR TITLE
Meeting phase1

### PR DESCRIPTION
**Pull request by andrewholsted**
## Description

This PR brings in the `meetings-plugin` and also the idea of a `StatelessSparkPlugin` to not user ampersand state. 

It's still very much a WIP but the basic capability to join a meeting via SIP URI works, and you can mute/unmute both audio and video. 

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Coverage

There are no tests. 

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* SDK Version
* Node/Browser Version
* NPM Version

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/roap/util.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+RoapUtil.updatePeerConnection = (meeting, session) =>
+  PeerConnectionManager.updatePeerConnection({
+    offerSdp: session.OFFER.sdps,
+    pc: meeting.mediaPeerConnection,
+    screenPeerConnection: meeting.screenPeerConnection,
+    offerToReceiveAudio: true,
+    offerToReceiveVideo: true
+  });
```
Comment (andrewholsted) [inline packages/node_modules/@webex/plugin-meetings/src/roap/util.js]: See about calling PCM directly

File: packages/node_modules/ciscospark/src/ciscospark.js [hunk 2]
```diff
@@ -9,9 +9,9 @@
 /* eslint camelcase: [0] */
 
 require('@ciscospark/plugin-authorization');
-require('@ciscospark/plugin-phone');
 // explicitly load wdm, since we're relying on preDiscoveryServices and the
 // url interceptor
+require('@ciscospark/internal-plugin-mercury');
```
Comment (andrewholsted) [inline packages/node_modules/ciscospark/src/ciscospark.js]: Import directly in the plugin-meeting

File: packages/node_modules/@webex/plugin-meetings/src/roap/index.js [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    Events.on(EVENTS.ROAP_ANSWER, this.sendRoapAnswer.bind(this));
+  }
+
+  /**
+   * Starts listening to mercury events for Roap messages
+   * @returns {Promise}
+   */
+  start() {
+    return Promise.try(() => {
+      this.spark.internal.mercury.on(ROAP.ROAP_MERCURY, (result) => {
```
[the remark this answers is not in the document]
Comment (andrewholsted) [reply packages/node_modules/@webex/plugin-meetings/src/roap/index.js]: Will tackle separately in a ROAP refactor branch

File: packages/node_modules/@webex/plugin-meetings/src/constants.js [hunk 4]
```diff
@@ -169,8 +183,8 @@ export const ROAP = {
   RECEIVE_CALL_LEAVE: 'RECEIVE_CALL_LEAVE',
   ROAP_MERCURY: 'event:locus.message.roap',
   ROAP_VERSION: '2',
-  RX_: 'RX_',
-  TX_: 'TX_'
+  RX: 'RX_',
+  TX: 'TX_'
```
Comment (andrewholsted) [inline packages/node_modules/@webex/plugin-meetings/src/constants.js]: Add lint rule to enable dangling

File: packages/node_modules/@webex/plugin-meetings/src/roap/handler.js [hunk 5]
```diff
@@ -119,16 +133,24 @@ export default class RoapHandler extends StatelessSparkPlugin {
...
     switch (action.type) {
       case ROAP.RECEIVE_ROAP_MSG:
         action.remote = true;
-        this.execute(signal, session, action, meeting, ROAP.RX_);
+        this.execute(signal, session, action, meeting, ROAP.RX);
         break;
       case ROAP.SEND_ROAP_MSG:
         action.local = true;
-        this.execute(signal, session, action, meeting, ROAP.TX_);
+        this.execute(signal, session, action, meeting, ROAP.TX);
```
Comment (andrewholsted) [inline packages/node_modules/@webex/plugin-meetings/src/roap/handler.js]: Update here after updating constant

File: packages/node_modules/@ciscospark/spark-core/src/lib/spark-core-plugin-mixin.js [hunk 6]
```diff
@@ -58,15 +58,17 @@ export default function mixinSparkCorePlugins(State, config, interceptors) {
...
-    if (constructor.prototype._definition.ready) {
-      const {fn, depList} = State.prototype._derived.ready;
-      const def = {
-        deps: depList.concat(`${name}.ready`),
-        fn
-      };
-
-      createDerivedProperty(State.prototype, 'ready', def);
+    // Only mess with the plugin's derived properties if it's an amp-state plugin
+    if (constructor.prototype._definition) {
```
Comment (mattnorris) [inline packages/node_modules/@ciscospark/spark-core/src/lib/spark-core-plugin-mixin.js]: Could this be written `if (constructor.prototype._definition && constructor.prototype._definition.ready)` ?

File: packages/node_modules/@ciscospark/spark-core/src/lib/spark-internal-core-plugin-mixin.js [hunk 7]
```diff
@@ -51,15 +51,17 @@ export default function mixinSparkInternalCorePlugins(State, config, interceptor
...
-    if (constructor.prototype._definition.ready) {
-      const {fn, depList} = State.prototype._derived.ready;
-      const def = {
-        deps: depList.concat(`${name}.ready`),
-        fn
-      };
-
-      createDerivedProperty(State.prototype, 'ready', def);
+    // Only mess with the plugin's derived properties if it's an amp-state plugin
+    if (constructor.prototype._definition) {
```
Comment (mattnorris) [inline packages/node_modules/@ciscospark/spark-core/src/lib/spark-internal-core-plugin-mixin.js]: See above.

File: packages/node_modules/@webex/plugin-meetings/README.md [hunk 8]
```diff
@@ -11,6 +11,9 @@
 - [Maintainers](#maintainers)
 - [License](#license)
 
+# WARNING: This plugin is currently under active development, is not stable, and breaking changes can and will happen!
+
+If you are looking to make calls through webex teams, `plugin-phone` is the package you are looking for...
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/README.md]: To create supported meetings and calls using Webex Teams, use [@ciscospark/plugin-phone](https://www.npmjs.com/package/@ciscospark/plugin-phone).

**Pull request merged**